src/client/Client.cc

```diff
@@ -1731,6 +1731,7 @@ int Client::make_request(MetaRequest *request,
     r = request->get_abort_code();
     request->item.remove_myself();
     unregister_request(request);
+    put_request(request);
```

Comment:
Doesn't `unregister_request` also do `put_request`?

Reply:
both mds_requests map and caller of make_reuqets() hold a reference. unregister_request() releases reference hold by mds_requests map

Reply:
my bad, introduced by 9cb79067dc009b488c9dc2d0c4641da88153bfca. it's a "potential" use-after-free.